terraform/provider/resource_teleport_github_connector.go

```diff
@@ -0,0 +1,174 @@
+		ReadContext:   resourceGithubConnectorRead,
+		UpdateContext: resourceGithubConnectorUpdate,
+		DeleteContext: resourceGithubConnectorDelete,
+
+		Schema:        tfschema.SchemaGithubConnectorV3,
+		SchemaVersion: 3,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// resourceGithubConnectorCreate creates Teleport GithubConnector from resource definition
+func resourceGithubConnectorCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c, err := getClient(m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	n, err := getResourceName(d, "github")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = c.GetGithubConnector(ctx, n, false)
+	if err == nil {
+		existErr := "GitHub connector " + n + " exists in Teleport. Either remove it (tctl rm github/" + n + ")" +
+			" or import it to the existing state (terraform import teleport_github_connector." + n + " " + n + ")"
+
+		return diag.FromErr(trace.Errorf(existErr))
+	}
+	if err != nil && !trace.IsNotFound(err) {
+		return diag.FromErr(describeErr(err, "role"))
+	}
+
+	cn := types.NewGithubConnector(n, types.GithubConnectorSpecV3{})
+
+	cn3, ok := cn.(*types.GithubConnectorV3)
+	if !ok {
+		return diag.FromErr(fmt.Errorf("failed to convert created role to types.GithubConnectorV3 from %T", cn))
+	}
+
+	err = tfschema.GetGithubConnectorV3(cn3, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = c.UpsertGithubConnector(ctx, cn3)
+	if err != nil {
+		return diag.FromErr(describeErr(err, "github"))
+	}
+
+	d.SetId(cn3.GetName())
+
+	return resourceGithubConnectorRead(ctx, d, m)
+}
+
+// resourceGithubConnectorRead reads Teleport role. This method is required by Terraform to ensure that CRUD
+// operation was successful.
+func resourceGithubConnectorRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c, err := getClient(m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := d.Id()
+
+	cn, err := c.GetGithubConnector(ctx, id, true)
+	if trace.IsNotFound(err) {
+		d.SetId("")
+		return diag.Diagnostics{}
+	}
+
+	if err != nil {
+		return diag.FromErr(describeErr(err, "github"))
+	}
+
+	cn3, ok := cn.(*types.GithubConnectorV3)
+	if !ok {
+		return diag.FromErr(fmt.Errorf("failed to convert created user to types.GithubConnectorV3 from %T", cn))
+	}
+
+	err = tfschema.SetGithubConnectorV3(cn3, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+// resourceGithubConnectorUpdate updates Teleport role from resource definition
+func resourceGithubConnectorUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c, err := getClient(m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := d.Id()
+
+	// This is the existence check. Since, there is no separate CreateRole method, we need to
+	// check that Role we are updating exists to avoid it's creation via UpsertRole.
```

Comment:
```suggestion
	// This is the existence check. Since, there is no separate CreateGithubConnector method, we need to
	// check that connector we are updating exists to avoid it's creation via UpsertGithubConnector.
```